app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -376,78 +399,86 @@ constructor(
                             event.tags.any { tag ->
                                 filterTagSet.any { tag2 -> tag.tagId == tag2.tagId }
                             } ||
                             areWordsInTitle(event, filterWordList) ||
                             areWordsInDescription(event, filterWordList)
                     }
-                    .filter { event -> // filter by time
+                    .filter { event ->
+                        // filter by time
                         dateFilterConditions(event)
                     }
                     .filter { event ->
+                        // filter by confirmed check
                         !_filtersContainer.value.confirmedChecked ||
                             event.maxParticipants <= 0 ||
                             (event.participantCount >= event.maxParticipants * STATUS_THRESHOLD &&
                                 event.participantCount < event.maxParticipants)
                     }
                     .filter { event ->
+                        // filter by pending check
                         !_filtersContainer.value.pendingChecked ||
                             event.maxParticipants <= 0 ||
                             event.participantCount < event.maxParticipants * STATUS_THRESHOLD
                     }
                     .filter { event ->
+                        // filter by full check
                         !_filtersContainer.value.fullChecked ||
                             event.maxParticipants <= 0 ||
                             event.participantCount == event.maxParticipants
                     }
                     .filter { event ->
+                        // filter by epfl check
                         !_filtersContainer.value.epflChecked ||
                             (!event.tags.any { tag -> sectionTags.contains(tag) } &&
                                 !event.tags.any { tag -> semesterTags.contains(tag) })
                     }
                     .filter { event ->
+                        // filter by section
                         !_filtersContainer.value.sectionChecked ||
                             _section.value == "" ||
                             event.tags.any { tag ->
                                 tag.name.lowercase() == _section.value.lowercase()
                             }
                     }
                     .filter { event ->
+                        // filter by semester
                         !_filtersContainer.value.classChecked ||
                             _semester.value == "" ||
                             event.tags.any { tag ->
                                 tag.name.lowercase() == _semester.value.lowercase()
                             }
                     }
+                    .filter { event ->
+                        // filter by association
+                        _selectedAssociation.value == -1 ||
```

Comment:
Knitpick: -1 can be defined as a private constant and then used here instead of directly using the value.